core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -3200,8 +3208,42 @@ private boolean o3Commit(long lag) {
             // Safe check of the sort. No known way to reproduce
             assert o3TimestampMin <= o3TimestampMax;
 
-            if (lag > 0) {
-                long lagThresholdTimestamp = o3TimestampMax - lag;
+            if (o3MaxLag > 0) {
+                long lagError = 0;
+                if (getMaxTimestamp() != Long.MIN_VALUE) {
+
+                    // When table already has data we can calculate the overlap of the newly added
+                    // batch of records with existing data in the table. Positive value of the overlap
+                    // means that our o3MaxLag was undersized.
```

Comment:
sounds like this should be:
"our o3EffectiveLag was undersized"

Reply:
fixed in https://github.com/questdb/questdb/pull/2759/commits/cd55db87ed385ca10a67463e0e120dcb4c10e123